### DISCUSSION TITLE
Linux framebuffer debugging on host machine (WSL)

### DISCUSSION THREAD

**Discussion by igiona** [unoplatform/uno#15510] [Q&A]
Hi

is it possible to debug the Linux Framebuffer head via WSL?
If yes, how?

If a simply run the debugger from within VisualStudio on WSL I get this error 

```shell
System.InvalidOperationException: 'Cannot read keys when either application does not have a console or when console input has been redirected. Try Console.Read.'
```

**Comment by jeromelaban** [comment]
Thanks for the question. 

As far as we know, it's not possible to run framebuffer apps in WSL, as there's no available framebuffer to render to. You'll need to run on an actual device.

**Reply by igiona** [reply]
Thank you for the quick feedback.

I have found an emulator https://sixpak.org/fbe/ which actually builds& runs after installing and configuring VcXsrv on windows and XFCE on WSL.
The question would be now how it is possible to tell the Uno app to use /tmp/fbe_buffer as "driver" I guess?

**Reply by jeromelaban** [reply]
Thanks. It seems that FBE only supports 8 bits color palette, a mode that Uno does not support at this time. Still, if you find a framebuffer emulator that works with 16 or 24bpp modes, you can specify the framebuffer using the `FRAMEBUFFER` environment variable.

As for the original error message, this particular mode is hard-coded here: https://github.com/unoplatform/uno/blob/a4d04acf6ada0e93d724f56c5f74062b5a69136d/src/Uno.UI.Runtime.Skia.Linux.FrameBuffer/FramebufferHost.cs#L79.

You _may_ still be able to debug the process by starting it using the terminal, then attach to the process instead of launching it.
